netsim/modules/lag.py

```diff
@@ -0,0 +1,75 @@
+import typing
+import netaddr
+
+from box import Box, BoxList
+from . import _Module
+from .. import data
+from ..utils import log
+from ..augment import devices
+
+#
+# Checks if 2 lists have the same elements, independent of order
+#
+def same_list(l1:BoxList,l2:BoxList) -> bool:
+  for l in l1:
+    if l not in l2:
+      return False
+  return len(l1)==len(l2)
+
+class LAG(_Module):
+
+  """
+  link_pre_transform: Create virtual LAG links
+  """
+  def link_pre_transform(self, link: Box, topology: Box) -> None:
+    if log.debug_active('vlan'):
+      print(f'LAG link_pre_transform for {link}')
+    if 'lag' in link and link.get('type',"")!="lag":
+
+      # Lookup virtual LAG link with same id between same pair of nodes, create if not existing
+      vlag = None
+      for l in topology.links:
+        if 'lag' in l and l.get('type',"")=="lag" and l.lag.id == link.lag.id \
+            and same_list(l.interfaces,link.interfaces):
+          vlag = l
+          break
+      
+      if vlag is None:
+        vlag = data.get_box(link)
+        vlag.type = "lag"
+        vlag.linkindex = len(topology.links) + 1
+        vlag._linkname = f"links[{vlag.linkindex}]"
+        vlag.interfaces = [ i for i in link.interfaces ] # Make a deep copy, could use a set?
```

Comment:
BoxList(link.interfaces) might also work. Or link.interfaces + []

Reply:
I wanted a set for easier order independent comparison - but that would require link.interfaces to become a set too

Reply:
link.interfaces could have been a set if there wouldn't have been an individual who insisted on having two interfaces of the same box connected to the same link 🤷‍♂️ We have to do quite some code modification to accommodate him.

Reply:
We could easily keep accommodating that individual by including a counter, as follows:
```
link.interfaces: { { "node": "n1" }, { "node": "n1", "_j": 2 } }
```